javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/GuavaBoundedCache.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.tooling;
+
+import com.google.common.cache.Cache;
+import io.opentelemetry.javaagent.instrumentation.api.BoundedCache;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+class GuavaBoundedCache<K, V> implements BoundedCache<K, V> {
+
+  private final Cache<K, V> delegate;
+
+  public GuavaBoundedCache(Cache<K, V> delegate) {
+    this.delegate = delegate;
+  }
+
+  @Override
+  public V get(K key, Function<? super K, ? extends V> mappingFunction) {
+    try {
+      return delegate.get(key, () -> mappingFunction.apply(key));
+    } catch (ExecutionException e) {
+      throw new BoundedCache.Exception(e);
+    }
```

Comment:
and then you can delete BoundedCache.Exception and the associated catch blocks
```suggestion
    try {
      return delegate.get(key, () -> mappingFunction.apply(key));
    } catch (ExecutionException e) {
      // this shouldn't happen because the java.util.function.Function doesn't declare any checked exceptions
      throw new IllegalStateException(e);
    }
```

Reply:
This is a good point.  I had only really noticed the contract through its signature and not the javadoc which does clearly say that it only throws if the value lookup throws.  🤘  I'll change it.